server/middlewares/checkUser.js

```diff
@@ -15,4 +15,36 @@ export const checkUserRole = (req, res,next) => {
         })
     }
     next();
-}
\ No newline at end of file
+}
+
+export const checkAdminAndAboveRole = (req, res,next) => {
+    if(!['admin', 'super admin'].includes(req.session.role)){
+        return res.status(401).json({
+            status: 'failure',
+            message: 'Sorry, you do not have the required priviledge'
+        })
+    }
+    next();
+}
+
+
+export const checkTeacherAndAboveTole = (req, res,next) => {
```

Comment:
Typo: checkTeacherAndAbove**Role**

Might be easier to name it: `checkTeacherPrivilege`, `checkAdminPrivilege` etc.

Reply:
Cool. Consider it done